authenticator.go

```diff
@@ -0,0 +1,25 @@
+package core
+
+import "crypto/tls"
+
+type Authenticator interface {
+	GetClientCertificate(service ServiceType, hostPort string) (*tls.Certificate, error)
```

Comment:
Am I right in saying that we never call this yet?

Reply:
Correct.